DOCS-908: Document part 1 of the cloud app API and move robot API (#1706)
viamrobotics/docs: docs/services/robot-service.md

linkTitle: "Robot"
description: "The Robot Service consists of a set of robot monitoring and management APIs that most robots support."
type: "docs"

> draft: "false"

Author: ```suggestion
draft: "true"
```

weight: 10
tags: ["robot state", "services"]
# SME: Cheuk
---

> Robot Service constitutes a minimal set of APIs that most robots (Viam Server, [Viam Python SDK](https://python.viam.dev/) and various SDKs) should support.

Reviewer: ```suggestion
The Robot Service refers to the minimum set of APIs that most robots running `viam-server` support.
This means that any of the various robot clients in Viam, like the Python SDK, Go Client SDK, or modular resources, should also support methods to interface with these same APIs from the client (instead of server) side.
If you are implementing your own client, keep this in mind.
```

Author: Oh oops I'm sorry-- I did not mean to include this file in this PR. I think it should stay as a draft and not be reviewed for now. My thought with this PR was to just move the method documentation that you already made on the APIs page to a different page for consistency, but not to address this page yet.

> Users will likely use the Robot Service as an entrypoint to interact with Viam robots and provide a way to get updates from the robot as a whole.

Reviewer: ```suggestion
Use the Robot Service as an entrypoint to interact with Viam robots and provide a way to get updates from the robot as a whole.
```
Needs to speak to users instead of at them. I'm not opposed to putting the robot service page back up with the current state of our docs, but would emphasize that this whole part was clearly written for internal use so needs to be reworded if we want to have it client-facing.

Author: Yeah totally; sorry didn't mean to include this file in this PR

> 

Reviewer: ```suggestion
... session management api, robot api, vision api, geometries/spatial integrations apis 
```
(opt) Could link with the shortcodes here to these methods tables. excited to see how this page turns out, agree it'd be a great value add!

Author: Will leave for a different PR/ticket

tags: ["robot state", "services"]
# SME: Cheuk
---

> 

Reviewer 2: should this still be in draft mode? and if not we need an icon for this page :)

Author: Yeah it should still be in draft mode; see comment below. I'd added a newline to it and temporarily un-drafted it when I was first working on the PR, and then forgot to remove it from the PR, so Sierra reviewed it and made some suggestions that I implemented but I didn't mean to actually look over this page at all in this PR.

> The Robot Service refers to the minimum set of APIs that most robots running `viam-server` support.

Reviewer 2: Is there a link to an interface or somethign that defines this set of APIs?

Reviewer 2: or is that why it's still in draft and we'll do that later?

Author: Yeah this doc should not be reviewed. The only changes made were suggestions Sierra made that I figured I might as well implement since they were good suggestions, but I didn't mean to bring this doc into this PR at all.
